### PR TITLE
Time

### PR DESCRIPTION
**Pull request by aleeusgr**

### CODE REVIEW

File: tests/time-props-vesting.test.ts [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+	})
+
+	it ("succeeds cancellation", async ({network, alice, bob, program}) => {
+		const optimize = false; // need to add it to the context
+		const compiledScript = program.compile(optimize);
+		const validatorHash = compiledScript.validatorHash;
+		const validatorAddress = Address.fromValidatorHash(validatorHash);
+
+		const adaQty = 10;
+		const duration = 1000000;
```
Comment (aleeusgr) [inline tests/time-props-vesting.test.ts:79]: these two lines duplicate, add to context or hardcode

Comment (aleeusgr) [reply tests/time-props-vesting.test.ts:79]: also duration can be obtained from slot number and used to explore edge cases also

File: tests/vesting-cancel.test.ts [hunk 2]
```diff
@@ -56,11 +56,14 @@ describe("a vesting contract: Cancel transaction", async () => {
 		const aliceUtxos = await alice.utxos;
```
Comment (aleeusgr) [inline tests/vesting-cancel.test.ts:56]: rm
